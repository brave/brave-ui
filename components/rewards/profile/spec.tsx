/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Profile from './index'

describe('Profile tests', () => {
  const baseComponent = (props?: object) => <Profile id='profile' {...props} />

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent())
      const assertion = wrapper.find('#profile').length
      expect(assertion).toBe(1)
    })
  })
})
