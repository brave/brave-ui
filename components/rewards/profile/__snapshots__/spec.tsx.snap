// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile tests basic tests matches the snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c1 {
  -webkit-flex-basis: 24px;
  -ms-flex-preferred-size: 24px;
  flex-basis: 24px;
  position: relative;
}

.c2 {
  border-radius: 50%;
  width: 24px;
  height: 24px;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  margin-top: -5px;
}

.c4 {
  margin-left: 10px;
}

.c5 {
  white-space: nowrap;
  font-family: Muli;
  font-size: 14px;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.29;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  color: #686978;
}

.c6 {
  white-space: nowrap;
  padding-left: 5px;
  color: #b8b9c4;
}

<div
  className="c0"
  id="profile"
>
  <div
    className="c1"
    type={undefined}
  >
    <img
      className="c2"
      src=""
      type={undefined}
    />
  </div>
  <div
    className="c3"
    type={undefined}
  >
    <div
      className="c4"
      type={undefined}
    >
      <span
        className="c5"
        type={undefined}
      />
      <span
        className="c6"
        type={undefined}
      />
    </div>
  </div>
</div>
`;
